gpflow/monitor.py

```diff
@@ -0,0 +1,245 @@
+            For large values of `period` the task will be less frequently run.
+        :param extra_keywords: pass additional keywords to be monitored.
+            If the parameter's name includes any of the extra keywords specified it
+            will be monitored as well. By default, parameters that match the `kernel` or
+            `likelihood` keyword are monitored.
+        """
+        super().__init__(log_dir, period)
+        self.model = model
+        self.max_size = max_size
+        self.keywords_to_monitor = ["kernel", "likelihood"] + extra_keywords
+
+    def run(self, **unused_kwargs):
+        for name, parameter in parameter_dict(self.model).items():
+            # check if the parameter name matches any of the specified keywords
+            if any(keyword in name for keyword in self.keywords_to_monitor):
+                name = name.lstrip(".")  # keys are prepended with a '.', which we strip
+                self._summarize_parameter(name, parameter)
+
+    def _summarize_parameter(self, name: str, value: Union[float, np.ndarray]):
+        """
+        :param name: identifier used in tensorboard
+        :param value: value to be stored in tensorboard
+        """
+        value = tf.reshape(value, (-1,))
+        for i in tf.range(tf.minimum(tf.size(value), self.max_size)):
+            tf.summary.scalar(f"{name}[{i}]", value[i], step=self.current_step)
+
+
+class ScalarToTensorBoard(ToTensorBoard):
+    """ Stores the returns value of a callback in a TensorBoard. """
+
+    def __init__(self, log_dir: str, callback: Callable[[], float], name: str, period: int = 1):
+        """
+        :param log_dir: directory in which to store the tensorboard files.
+            For example, './logs/my_run/'.
+        :param callback: callback to be executed and result written to TensorBoard.
+            A callback can have arguments (e.g. data) passed to the function using
+            keyword arguments.
+            For example:
+            ```
+            lambda cb(x=None): 2 * x
+            task = ScalarToTensorBoard(logdir, cb, "callback")
+
+            # specify the argument of the function using kwargs, the names need to match.
+            task(step, x=1)
+            ```
+        :param name: name used in TensorBoard.
+        :param period: defines how often to run the task; it will execute every `period`th step.
+            For large values of `period` the task will be less frequently run.
+        """
+        super().__init__(log_dir, period)
+        self.name = name
+        self.callback = callback
+
+    def run(self, **kwargs):
+        tf.summary.scalar(self.name, self.callback(**kwargs), step=self.current_step)
+
+
+class ImageToTensorBoard(ToTensorBoard):
+    def __init__(
+        self,
+        log_dir: str,
+        plotting_function: Callable[[Figure, Axes], Figure],
+        name: Optional[str] = None,
+        period: int = 1,
+        *,
```

Comment:
```suggestion
        name: Optional[str] = None,
        *,
        period: int = 1,
```
With this many arguments calling `ImageToTensorBoard("./logs/my_run", plotting_callback, "my_plot", 5)` already seems pretty unintelligible - forcing people to explicitly write `period=5` could make this clearer. (Or `interval=5` :wink: )